### DISCUSSION TITLE
Tips for running smoothly on Apple Silicon?

### DISCUSSION THREAD

**Discussion by connorjewiss** [hacksider/Deep-Live-Cam#312] [Q&A]
Followed the GPU Acceleration tips for running smoothly of Apple Silicon, but still laggy compared to Windows versions I've seen. Running on 32GB M1 Max.

**Comment by sporkman** [comment]
Same here, I'm at around 2 fps, what should I be seeing?

**Comment by briascoi** [comment]
Same here, and no one is answering this question. There is no help at all with Mac :(

**Reply by hacksider** [reply]
Sorry about that, I don't really have a mac to begin with so I can't answer it. And having a dedicated GPU is really the best way to run this with CUDA ☺️

**Comment by circlesmiler** [comment]
Ok... I make a wild guess: You need a lot of GPU power to run this smoothly. Apple Silicon Chips  has integrated GPU which is OK for doing your work and playing some games. But there is a reason why gaming is still done a lot on PC + crazy graphic cards that have a power consumption like a hair dryer. :-) (PS: I'm a happy Mac user, but there are limitations)

**Reply by hacksider** [reply]
This, plus I don't have a mac to actually test it 😅

**Comment by cdrake118** [comment]
I think it is a bug because it utilizes 0% of GPU currently.

**Comment by hacksider** [comment]
I don't really have a mac to begin with to resolve the issue so we're at the mercy of the community help here. Plus there's really an advantage of having dedicated GPU.

**Reply by fakeneuron** [reply]
When looking at Activity Monitor, none of the work is being done by the GPU which runs at 0%. All is being done with CPU.

**Comment by manINsano-GIT** [comment]
same issue. looking for community support.

**Comment by dodgyrabbit** [comment]
The biggest improvement on mac (tested on M2 pro) comes from this:

Delete models/inswapper_128_fp16.onnx

Download this model instead:
https://huggingface.co/hacksider/deep-live-cam/blob/main/inswapper_128.onnx

Copy it to the models directory and rename it to **inswapper_128_fp16.onnx**

Then make sure you start with **coreml** execution provider, like so:
```
python run.py --execution-provider coreml
```

Should get around 4 fps

Note that this does *not* help you with the face_enhancer. That is a completely separate python module with it's own model and is not optimized for mac GPU. I believe it's all on CPU so you're going to get about 1 frame every few seconds if you turn it on.

**Reply by UltraSeeder** [reply]
Works good

**Reply by tomkirk** [reply]
this does improve it a bit thanks.

**Comment by tomkirk** [comment]
I also have this issue on an M3 pro. 

- 11-core CPU
- 14-core GPU

I am also experiencing about 2fps. 

Any further ideas on how to improve this?

**Comment by fakeneuron** [comment]
I have the same issue with M3 Macbook Air. When looking at Activity Monitor, i see all the work is being done by the CPU. GPU is at 0%. Looks like something in the code is not properly directing it to run on the GPU.

**Comment by awu0403** [comment]
Is this problem still unresolved at present? When looking at Activity Monitor, none of the work is being done by the GPU which runs at 0%. All is being done with CPU
